From 0d9d87c3af2d2d37cbddda0865c443da8dc6d056 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Tue, 27 Aug 2013 16:22:53 +0200
Subject: [PATCH] Disable the JIT compiler

It's causing problems on ARM platforms and let the WebProcess die with a SIGILL signal:

Program received signal SIGILL, Illegal instruction.
[Switching to LWP 657]
0x413e7aba in llint_slow_path_new_object () from /usr/lib/libQt5WebKit.so.5
(gdb) bt
0  0x413e7aba in llint_slow_path_new_object ()
   from /usr/lib/libQt5WebKit.so.5
1  0x413f1ab8 in llint_op_new_object () from /usr/lib/libQt5WebKit.so.5
2  0x413f1ab8 in llint_op_new_object () from /usr/lib/libQt5WebKit.so.5
Backtrace stopped: previous frame identical to this frame (corrupt stack?)

Upstream-Status: Inappropiate [disable feature]

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 Tools/qmake/mkspecs/features/default_post.prf | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Tools/qmake/mkspecs/features/default_post.prf b/Tools/qmake/mkspecs/features/default_post.prf
index a92aaa7..40048c9 100644
--- a/Tools/qmake/mkspecs/features/default_post.prf
+++ b/Tools/qmake/mkspecs/features/default_post.prf
@@ -42,6 +42,9 @@ root_project_file: finalizeConfigure()
 DEFINES += \
     BUILDING_QT__=1
 
+# Disabling JIT
+DEFINES += ENABLE_JIT=0
+
 CONFIG(release, debug|release): DEFINES *= NDEBUG
 
 DEFINES += $$configDefines()
-- 
1.8.1.2

